server/plugin/command.go

```diff
@@ -453,6 +459,151 @@ func (p *Plugin) handleIssue(_ *plugin.Context, args *model.CommandArgs, paramet
 	}
 }
 
+func (p *Plugin) handlewebhook(_ *plugin.Context, args *model.CommandArgs, parameters []string, userInfo *GitHubUserInfo) string {
+	if len(parameters) == 0 {
+		return "Invalid webhook command. Available command are `add` and `list` ."
+	}
+	command := parameters[0]
+	parameters = parameters[1:]
+	ctx := context.Background()
+	githubClient := p.getGithubClient(userInfo)
+	baseURL := p.getBaseURL()
+	switch command {
+	case add:
+		if len(parameters) < 1 {
+			return "Unknown action, Currently supported to add and List webhook "
+		}
+		owner, repo := parseOwnerAndRepo(parameters[0], p.getBaseURL())
+		if owner == "" {
+			return "Currently supported to add webhook to a specific [owner/repo] ."
+		}
+
+		siteURL := *p.API.GetConfig().ServiceSettings.SiteURL + inboundWebhookURL
+		var hook github.Hook
+		var config = make(map[string]interface{})
+
+		if len(parameters) > 2 {
+			for _, val := range strings.Split(parameters[2], ",") {
+				KVPair := strings.Split(val, "=")
+				switch {
+				case KVPair[0] == "insecure_ssl":
+					insecureSSL, err := strconv.ParseBool(KVPair[1])
+					if err != nil {
+						return "config filed insecure_ssl can only have on of the following true or false "
+					}
+					config[KVPair[0]] = insecureSSL
+				case KVPair[0] == "options":
+					hook.Events = strings.Split(KVPair[1], "&")
+				default:
+					config[KVPair[0]] = KVPair[1]
+				}
+			}
+		} else {
+			config["url"] = siteURL
+			config["insecure_ssl"] = false
+			config["content_type"] = "application/json"
+			hook.Events = []string{"*"}
+		}
+		hook.Config = config
+
+		githubHook, _, err := p.CreateHook(ctx, githubClient, owner, repo, hook)
+		if err != nil {
+			return err.Error()
+		}
+
+		hookDetails, _, err := p.GetHook(ctx, githubClient, owner, repo, *githubHook.ID)
+		if err != nil {
+			return err.Error()
+		}
+
+		hookURL := baseURL + owner
+		if repo != "" {
+			hookURL += "/" + repo
+		}
+		hookURL += "/settings/hooks/"
+		txt := "Webhook Created Successfully \n"
+		hookID := strconv.Itoa(int(*githubHook.ID))
+		txt += fmt.Sprintf(" *  [%s](%s) :  -  %s", hookID, hookURL+hookID, strings.Join(hookDetails.Events, " , "))
+		txt += newLine
+
+		return txt
+	case list:
+
+		if len(parameters) != 1 {
+			return "Unknown action, please use `/github help` to see all actions available."
+		}
+
+		owner, repo := parseOwnerAndRepo(parameters[0], p.getBaseURL())
+		var Page int
+		var ShouldCallNext = true
+		var githubHooks []*github.Hook
+		var githubResponse *github.Response
+		var err error
+		if repo == "" {
+			for ShouldCallNext {
```

Comment:
That's not the idiomatic way of doing paging. Any objections on following https://github.com/google/go-github#pagination?

Reply:
Sure will implement this.

for this conversonation https://github.com/mattermost/mattermost-plugin-github/pull/456#discussion_r695580234
and https://github.com/mattermost/mattermost-plugin-github/pull/456#discussion_r695580073 
implemented slash command